src/Components/Blazor/Blazor/src/Hosting/EntrypointInvoker.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Blazor.Hosting
+{
+    internal static class EntrypointInvoker
+    {
+        public static Task InvokeEntrypointAsync(IntPtr entrypointMethodHandleValue, string[] args)
+        {
+            var entrypointMethodBase = CreateMethodBase(entrypointMethodHandleValue);
+
+            // For "async Task Main", the C# compiler generates a method called "<Main>"
+            // that is marked as the assembly entrypoint. Detect this case, and instead of
+            // calling "<Whatever>", call the sibling "Whatever".
```

Comment:
What's the reason why we need to call `Main` instead of `<Main`> ? Is this because `<Main>` (generated wrapper) contains a `Task.GetAwaiter.GetResult()`?

Reply:
Precisely